server/src/main/java/org/opensearch/action/search/CreatePitController.java

```diff
@@ -258,23 +258,29 @@ public void onFailure(final Exception e) {
     /**
      * Cleanup all created PIT contexts in case of failure
      */
-    private void cleanupContexts(Collection<SearchContextIdForNode> contexts) {
-        ActionListener<Integer> deleteListener = new ActionListener<>() {
+    private void cleanupContexts(Collection<SearchContextIdForNode> contexts, String pitId) {
+        ActionListener<DeletePitResponse> deleteListener = new ActionListener<>() {
             @Override
-            public void onResponse(Integer freed) {
-                // log the number of freed contexts - this is invoke and forget call
-                logger.debug(() -> new ParameterizedMessage("Cleared contexts in {} nodes out of {}", freed, contexts.size()));
+            public void onResponse(DeletePitResponse response) {
+                // this is invoke and forget call
```

Comment:
Minor, since nothing except debug logging in done here, we may just short-circuit on `logger.isDebugEnabled()` and skip the for-each loop.